client-guide/jmap-client-guide.mdown

```diff
@@ -11,278 +9,294 @@ A single login may have access to multiple accounts. In this guide I'm going to
 
-The Inbox, Sent and Awaiting Reply mailboxes all have `mustBeOnlyMailbox:false`. The Trash mailbox on the other hand has `mustBeOnlyMailbox:true`. This means that a message in the trash may not be in any other mailbox, and vice versa. However, a message belonging to the Sent mailbox could also be in, for example, the Awaiting Reply mailbox.
-
 The Sent mailbox has no unread messages, but 2 unread threads. This is not an error! A thread is considered unread if any of the messages in it are unread, even if those messages are actually in a different mailbox. In this case, the 3 unread messages in the Inbox must all be replies to messages in the Sent mailbox. Email clients will probably want to hide unread thread counts for mailboxes with a role of "sent" or "archive".
 
-Presuming the client defaults to showing the `role=inbox` mailbox, we now get the list of messages at the top of the mailbox, and the data for each one that we need to show it in the interface.
-
-    [
-        [ "getMessageList", {
-            filter: {
-                inMailboxes: [ "mailbox1" ]
-            },
-            sort: [ "date desc", "id desc" ]
-            collapseThreads: true,
-            position: 0,
-            limit: 10,
-            fetchThreads: true,
-            fetchMessages: true,
-            fetchMessageProperties: [
-                "threadId",
-                "mailboxId",
-                "isUnread",
-                "isFlagged",
-                "isAnswered",
-                "isDraft",
-                "hasAttachment",
-                "from",
-                "to",
-                "subject",
-                "date",
-                "preview"
-            ],
-            fetchSearchSnippets: false
-        }, "call1"]
-    ]
+Presuming the client defaults to showing the `role=inbox` mailbox, we now use a chain of method calls to get all the data we need to display the messages at the top of the mailbox and the threads they participate in.
+
+First we do a query for the `id` of first 10 messages in the mailbox, sorted descending by received date:
+
+```json
+[
+    // First we do a query for the id of first 10 messages in the mailbox
+    [ "Email/query", {
+        "accountId": "u123456",
+        "filter": {
+            "inMailbox": "mailbox1"
+        },
+        "sort": [
+            { "property": "receivedAt", "isAscending": false }
+        ],
+        "collapseThreads": true,
+        "limit": 10
+    }, "0" ],
+    
+    // Then we fetch the threadId of each of those messages
+    [ "Email/get", {
+        "accountId": "u123456",
+        "#ids": {
+            "name": "Email/query",
+            "path": "/ids",
+            "resultOf": "0"
+        },
+        "properties": [ "threadId" ]
```

Comment:
Also, let's add a `"position": 0` argument. It's optional (because 0 is the default) but we talk about it lower down so it will be clearer if it's explicitly included.